### ISSUE TITLE
Crash when navigating into a submodule

### ISSUE THREAD

**Issue by Zykino** [extrawurst/gitui#1590]
**Describe the bug**
Crash or hang when entering a submodule (the only crash logs I got is below). Most of the time it just open the submodule, wait a bit, and then go back to the shell as if gitui closed. But the shell still is waiting on gitui. Entering arrows give `^[[C^[[D^[[A^[[B`, I need to `Ctr+c` to stop gitui.

**To Reproduce**
1. `Shift + S`
2. `Enter` on any listed submodule
3. Wait for 1 or 2 second without doing anything.

**Expected behavior**
A gitui view of the submodule (and being able to commit?).

**Context (please complete the following information):**
 - OS/Distro + Version: uname -av: `Linux PF2WDF9K 5.4.72-microsoft-standard-WSL2 #1 SMP Wed Oct 28 23:40:43 UTC 2020 x86_64 GNU/Linux`
 - GitUI Version: `0.22.1`
 - Rust version: `1.67.1`? Not sure when was the last time `cargo install-update -a` updated gitui, nor the rust version at the time...
 - WindowsTerminal version: `1.15.2874.0`
 - Zellij version: `0.34.4` (got errors even out of it)
 - fish, version 3.0.2 (for completness but also got it with GNU bash, version 5.0.3(1)-release (x86_64-pc-linux-gnu))

I suspect WSL/Windows or WindowsTerminal.

Also my submodules are in a different host than the principal repo (think developping from github but having submodules in gitlab). It kinds of is another issue, but it may be relevant: I cannot pull/push directly from gitui since I am behind a proxy. I configured git with this commands (and it works for the git cli).
```sh
git config --global http.http://<git-server-url>:<port>.proxy http://<proxy-url>:<port>
git config --global https.https://<git-submodules-server-url>:<port>.proxy http://<proxy-url>:<port>
``` 

**Additional context**
Crash log
```sh
❯ gitui
panic: Any { .. }
trace:
   0: gitui::set_panic_handlers::{{closure}}
   1: <rayon_core::job::HeapJob<BODY> as rayon_core::job::Job>::execute
   2: std::sys_common::backtrace::__rust_begin_short_backtrace
   3: core::ops::function::FnOnce::call_once{{vtable.shim}}
   4: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/alloc/src/boxed.rs:1940:9
      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/alloc/src/boxed.rs:1940:9
      std::sys::unix::thread::Thread::new::thread_start
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/sys/unix/thread.rs:108:17
   5: start_thread
             at /build/glibc-6iIyft/glibc-2.28/nptl/pthread_create.c:486:8
   6: clone
             at /build/glibc-6iIyft/glibc-2.28/misc/../sysdeps/unix/sysv/linux/x86_64/clone.S:95

fish: “gitui” terminated by signal SIGABRT (Abort)
```
The exit code was 134

**Comment by extrawurst** [comment]
Can you build from master and test again? A few improvements did not get released yet on this

**Comment by Zykino** [comment]
Yes `gutui` built from source does not crash when entering a submodule. Will the next version be deployed soon or should I expect to stay on the git repo for some time?

I’m not sure if it should be another issue, but when leaving a submodule I wanted to get back to the original repo:
- In main repo, see I modified files in a submodule
- Enter the submodule, commit & push or discard the changes
- Go back to the main repo to commit & push there.

Maybe I’m just leaving gitui (with `q` or `ESC`) when I should use another hotkey to getback up one level?

**Comment by extrawurst** [comment]
@Zykino you actually have to open the submodule popup again to jump up into the parent repo

**Comment by extrawurst** [comment]
next release is soon

**Comment by extrawurst** [comment]
closing this as not relevant anymore

**Comment by MaiLunJiye** [comment]
UI crash when navigating into submodule:

gitui version: 0.22.1
system: 
- Mac Ventura  13.2 (22D49)
- Debian 9 (connect via ssh)

reproduce enviroment

```
git clone https://github.com/ava-labs/avalanchego.git
cd avalanchego

# simulated submodule
git submodule add https://github.com/ava-labs/avalanchego.git script/sub
```

reproduce step : 

1. cd to `avalanchego`
2. Use Shift + S navigate to submodule

Then ui will crash. The terminal cursor will disappear

**Comment by Zykino** [comment]
Did you read the comments and tried the version from master branch directly?
